precompiles/assets/testdata/Gateway.sol

```diff
@@ -0,0 +1,65 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.17;
+
+import "../IAssets.sol";
+
+contract Gateway {
+    // Deposit LST
+    function depositLST(
+        uint32 clientChainID,
+        bytes calldata assetsAddress,
+        bytes calldata stakerAddress,
+        uint256 opAmount
+    ) public returns (bool success, uint256 latestAssetState) {
+        // Call the precompile
+        (success, latestAssetState) = ASSETS_CONTRACT.depositLST(
+            clientChainID,
+            assetsAddress,
+            stakerAddress,
+            opAmount
+        );
+
+        return (success, latestAssetState);
+    }
+
+    // Withdraw LST
+    function withdrawLST(
+        uint32 clientChainID,
+        bytes calldata assetsAddress,
+        bytes calldata withdrawAddress,
+        uint256 opAmount
+    ) public returns (bool success, uint256 latestAssetState) {
+        // Call the precompile
+        (success, latestAssetState) = ASSETS_CONTRACT.withdrawLST(
+            clientChainID,
+            assetsAddress,
+            withdrawAddress,
+            opAmount
+        );
+
+        return (success, latestAssetState);
+    }
+
+    function withdrawLSTAndThenRevert(
+        uint32 clientChainID,
+        bytes calldata assetsAddress,
+        bytes calldata withdrawAddress,
+        uint256 opAmount
+    ) public returns (bool success, uint256 latestAssetState) {
+        (success, latestAssetState) = withdrawLST(clientChainID, assetsAddress, withdrawAddress, opAmount);
+        revert("Intentional revert after withdraw");
```

Comment:
The scenario that reflects the problem this time is not a direct revert following the precompile call, but calling another contract after calling the precompile and revert in this outcome call. So instead of directly reverting here, we should call another contract here and make that call revert

Reply:
Gateway is not called in the unit test. Instead, GatewayCaller calls gateway.

Reply:
Yeah, I mean the old node would not trigger the error if we directly revert after calling the precompile in the same function, unless we call another contract and revert there, according to my test

Reply:
Updated so that a callee contract is responsible for the revert instead.